### DISCUSSION TITLE
.git folder bloat

### DISCUSSION THREAD

**Discussion by Kalekki** [zero01101/openOutpaint#38] [General]
So i noticed while doing git clone that it now downloads +148mb(!!!) which is odd since the massive gifs were removed. 
![image](https://user-images.githubusercontent.com/1765167/203585031-bed345c3-cd08-4b8a-82fe-86ec47344fbf.png)

![image](https://user-images.githubusercontent.com/1765167/203583778-531eddc3-fb81-4605-b38b-bf82f8d0771a.png)
How do we get rid of this bloat?

**Comment by SMUsamaShah** [comment]
It's big because all those files are still in git history.
@zero01101 will have to rewrite the history such that only last gif is there. https://stackoverflow.com/questions/43762338/how-to-remove-file-from-git-history

Was going to suggest squashing the commits but can't squash all the way down to first gif.

**Comment by zero01101** [comment]
D: D: oh holy crap yeah that's not good at all, will 100% run some git filter-branch on this after some PR merges, my apologies!

ps wiztree is fantastc, great choice

**Reply by SMUsamaShah** [reply]
I like Space Sniffer.

**Reply by zero01101** [reply]
spacesniffer's a close second, wiztree's just so _fast_

**Comment by zero01101** [comment]
:/ well `git filter-branch --tree-filter 'rm -f *.gif' HEAD` sure appeared to be Doing Things, so once it finished i was apparently an equal number of commits ahead and behind origin which i... guess makes sense?  stack overflow etc all kind of led me to force push the changes, but that seems to have done nothing for the repo size...

```
$ git count-objects -v
size: 153143
in-pack: 8
packs: 1
size-pack: 14
prune-packable: 0
garbage: 0
size-garbage: 0

```

**Reply by seijihariki** [reply]
Do you think rewriting history could be a possibility? As stated on the first answer here:

https://stackoverflow.com/questions/29042783/how-can-i-delete-all-commits-before-a-given-date-in-git-history

As we aren't really doing this to save space in the git server, we can save the full history on a old-main branch and modify the current main to cut off the gif files from the history.

Still, if you think that's too radical and want to avoid doing that, I understand.

**Reply by zero01101** [reply]
i'm open to pretty much anything lol, i've got a file-level backup archive anyway so if i burn it all down i should _hopefully_ be able to un-burn it all down... in theory

please see previous mentions of i'm an idiot etc :D

anyway,
>Do you think rewriting history could be a possibility?

i genuinely thought that was precisely what i was doing, based off of the section titled ["Removing a File from Every Commit"](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) but smart money's on the likelihood that i have once again Done It Wrong :)

changing main to old-main sounds like a great idea, i mean i think i've written the least code out of everyone now lol

can't guarantee i'll get this done, uh, immediately; holidays coming up here in the US kinda demanding some extra attention, and again, idiocy etc lol

**Reply by zero01101** [reply]
well i sure tried this again and now i have another branch that's seemingly identical to main all the way back, brilliantly titled "new-main" so that's awesome lol

**Comment by zero01101** [comment]
ok i think i finally have this fixed now, new clone is ~40mb :)
